packages/react-query/src/useSuspenseInfiniteQuery.ts

```diff
@@ -41,5 +43,5 @@ export function useSuspenseInfiniteQuery<
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
     InfiniteQueryObserver as typeof QueryObserver,
     queryClient,
-  ) as DefinedUseInfiniteQueryResult<TData, TError>
+  ) as InfiniteQueryObserverSuccessResult<TData, TError>
```

Comment:
I think alternative solution is below suggestion

```suggestion
  ) as unknown as UseSuspenseInfiniteQueryResult<TData, TError>
```